velox/functions/lib/window/RowNumber.cpp

```diff
@@ -19,13 +19,15 @@
 #include "velox/expression/FunctionSignature.h"
 #include "velox/vector/FlatVector.h"
 
-namespace facebook::velox::window::prestosql {
+namespace facebook::velox::functions::window {
 
 namespace {
 
+template <typename T>
```

Comment:
We don't need to templatize the entire class. Abstract a function that does 

template <typename T>
void applyBlock(...) {
  int numRows = peerGroupStarts->size() / sizeof(vector_size_t);
  auto* rawValues = result->asFlatVector<T>()->mutableRawValues();
    for (int i = 0; i < numRows; i++) {
      rawValues[resultOffset + i] = rowNumber_++;
    }
}

And just call the applyBlock in apply based on the result type.

Reply:
Done.